sdk/core/Azure.Core/src/AzureExtensions.cs

```diff
@@ -1,17 +0,0 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
-using System.Threading;
-using System.Threading.Tasks;
-
-namespace Azure
-{
-    public static class AzureExtensions
-    {
-        public static async ValueTask<Response<T>> WaitCompletionAsync<T>(this Task<Operation<T>> operation, CancellationToken cancellationToken = default) where T : notnull
```

Comment:
I'm not sure it having this method is worth it. Having extension methods on Task<T> is an uncommon pattern and people are more used to awaiting. It also allows us to remove AzureExtension type that we're not happy with the name of.

Reply:
Without this, people cannot simply await the LRO method call. I would be very concerned with removing it so late in the product cycle after doing usability studies and observing people calling this API.

Reply:
Did anyone use this call in the user study?

> Without this, people cannot simply await the LRO method call. 

This method doesn't work with `Operation<T>` subclasses anyway.

Reply:
Fixed

Reply:
Yeah, we had people find this and use it in our UX studies as well. They commented positively on it.